Flaky Test TestSlotMigrateFromSlave

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.


### Motivation

```
--- FAIL: TestSlotMigrateFromSlave (2.36s)
    server.go:106: 
        	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/server.go:106
        	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/server.go:112
        	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/server.go:101
        	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:47
        	            				/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:65
        	Error:      	Received unexpected error:
        	            	signal: aborted (core dumped)
        	Test:       	TestSlotMigrateFromSlave
FAIL
```

### Solution

Perhaps we should upload server output on test failures? cc @PragmaTwice 

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

@git-hulk they're somehow different issues - not the exact failure cause, or not the exact test case:

```
--- FAIL: TestSlotMigrateDataType (85.48s)
    --- FAIL: TestSlotMigrateDataType/MIGRATE_-_Migrate_incremental_data_via_parsing_and_filtering_data_in_WAL (60.85s)
        slotmigrate_test.go:703: 
            	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:703
            	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:625
            	Error:      	Condition never satisfied
            	Test:       	TestSlotMigrateDataType/MIGRATE_-_Migrate_incremental_data_via_parsing_and_filtering_data_in_WAL
FAIL
exit status 1
```

```
--- FAIL: TestPubSub (1.20s)
    server.go:106: 
        	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/server.go:106
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/server.go:112
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/server.go:101
        	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/pubsub_test.go:369
        	Error:      	Received unexpected error:
        	            	signal: abort trap
        	Test:       	TestPubSub
FAIL
exit status 1
```

> @git-hulk they're somehow different issues - not the exact failure cause, or not the exact test case:
> 
> ```
> --- FAIL: TestSlotMigrateDataType (85.48s)
>     --- FAIL: TestSlotMigrateDataType/MIGRATE_-_Migrate_incremental_data_via_parsing_and_filtering_data_in_WAL (60.85s)
>         slotmigrate_test.go:703: 
>             	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:703
>             	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:625
>             	Error:      	Condition never satisfied
>             	Test:       	TestSlotMigrateDataType/MIGRATE_-_Migrate_incremental_data_via_parsing_and_filtering_data_in_WAL
> FAIL
> exit status 1
> ```
> 
> ```
> --- FAIL: TestPubSub (1.20s)
>     server.go:106: 
>         	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/server.go:106
>         	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/server.go:112
>         	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/server.go:101
>         	            				/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/pubsub/pubsub_test.go:369
>         	Error:      	Received unexpected error:
>         	            	signal: abort trap
>         	Test:       	TestPubSub
> FAIL
> exit status 1
> ```

Yes, for the `TestPusSub` test case, it should be the same with [TestSlotMigrateFromSlave](https://github.com/apache/incubator-kvrocks/issues/1016#top), there're core when exiting the server.

--- a/src/cluster/replication.cc
+++ b/src/cluster/replication.cc
@@ -933,6 +933,7 @@ bool ReplicationThread::isWrongPsyncNum(const char *err) {
 
 rocksdb::Status WriteBatchHandler::PutCF(uint32_t column_family_id, const rocksdb::Slice &key,
                                          const rocksdb::Slice &value) {
+  type_ = kBatchTypeNone;
   if (column_family_id == kColumnFamilyIDPubSub) {
     type_ = kBatchTypePublish;
     kv_ = std::make_pair(key.ToString(), value.ToString());

--- a/src/cluster/replication.h
+++ b/src/cluster/replication.h
@@ -49,7 +49,8 @@ enum ReplState {
 };
 
 enum WriteBatchType {
-  kBatchTypePublish = 1,
+  kBatchTypeNone = 0,
+  kBatchTypePublish,
   kBatchTypePropagate,
   kBatchTypeStream,
 };
@@ -217,5 +218,5 @@ class WriteBatchHandler : public rocksdb::WriteBatch::Handler {
 
  private:
   std::pair<std::string, std::string> kv_;
-  WriteBatchType type_;
+  WriteBatchType type_ = kBatchTypeNone;
 };

